docs/source/tools/trigger-service.rst

```diff
@@ -0,0 +1,113 @@
+
+Complex workflows can require running many triggers for many parties and at a certain point, use of ``daml trigger`` with its process per trigger model becomes unwieldy. The Trigger Service provides the means to host multiple triggers for multiple parties running against a common ledger in a single process and provides a convenient interface for starting, stopping and monitoring them.
+
+The trigger-service is a ledger client that acts as an end-user agent. The trigger service intermediates between the ledger and end-users by running triggers on their behalf. The trigger-service is an HTTP REST service. All requests and responses use JSON to encode data.
+
+Starting the Trigger Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, it is assumed there is a sandbox ledger running on port 6865 on localhost.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key
+
+The above starts the Trigger Service using a number of default parameters. Most notably, the HTTP port the Trigger Service listens on which defaults to 8088. The meanining of the ``--no-secret-key`` parameter will be explained in a later section. To see all of the available parameters, their defaults and descriptions, one can execute the command ``daml trigger-service --help``.
+
+Although as we'll see, the Trigger Service exposes an endpoint for end-users to upload DAR files to the service it is sometimes convenient to start the service preconfigured with a specific DAR. To do this, the ``--dar`` option is provided.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key --dar <DAR>
+
+End-user interaction with the Trigger Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``start``
+*********
+
+Start a trigger. In this example, Alice starts the trigger called ``trigger`` in a module called ``TestTrigger`` of a package with ID "312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14". The response contains an identifier for the running trigger that Alice can use in subsequent commands involving the trigger.
+
+.. code-block:: bash
+
+   $ curl --user 'alice:&alC2l3SDS*V' \
+       -X POST localhost:8088/v1/start \
+       -H "Content-type: application/json" -H "Accept: application/json" \
+       -d '{"triggerName":"312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14:TestTrigger:trigger"}'
+   {"result":{"triggerId":"4d539e9c-b962-4762-be71-40a5c97a47a6"},"status":200}
+
+``stop``
+********
+
+Stop a running trigger. Alice stops her running trigger like so.
+
+.. code-block:: bash
+
+   $curl --user 'alice:&alC2l3SDS*V' \
+      -X DELETE localhost:8088/v1/stop/4d539e9c-b962-4762-be71-40a5c97a47a6 \
```

Comment:
Unrelated to this PR: The fact that you send a `DELETE` to a `stop` endpoint feels wrong. I think the REST-ful thing would be to send delete to a generic endpoint for the trigger, e.g., `localhost:8088/v1/triggerid`

Reply:
i'll get this in an issue too.

Reply:
https://github.com/digital-asset/daml/issues/6333